File: Java_Programs/src/com/example/programs/Program_3.java
package com.example.programs;

import java.util.ArrayList;
import java.util.List;

public class Program_3 {
    public static int[] rearrangeArray(int[] arr) {
        List<Integer> pos = new ArrayList<>();
        List<Integer> neg = new ArrayList<>();
        for (int num : arr) {
            if (num >= 0) {
                pos.add(num);
            } else {
                neg.add(num);
            }
        }
        int[] res = new int[arr.length];
        int i = 0;
        while (!pos.isEmpty() && !neg.isEmpty()) {
            res[i++] = neg.remove(0);
            res[i++] = pos.remove(0);
        }
        while (!pos.isEmpty()) {
            res[i++] = pos.remove(0);
        }
        while (!neg.isEmpty()) {
            res[i++] = neg.remove(0);
        }
        return res;
    }
}

File: Python_Programs/src/Program_3.py
def rearrange_array(arr):
    n = len(arr)
    for i in range(n):
        if i % 2 == 0 and arr[i] >= 0:
            for j in range(i + 1, n):
                if arr[j] < 0:
                    arr[i:j+1] = [arr[j]] + arr[i:j]
                    break
        elif i % 2 == 1 and arr[i] < 0:
            for j in range(i + 1, n):
                if arr[j] >= 0:
                    arr[i:j+1] = [arr[j]] + arr[i:j]
                    break
    return arr

'''
def rearrange_array(arr):
    n = len(arr)
    pos, neg = [], []
    for num in arr:
        if num >= 0:
            pos.append(num)
        else:
            neg.append(num)
    res = []
    while pos and neg:
        res.append(neg.pop(0))
        res.append(pos.pop(0))
    res.extend(pos)
    res.extend(neg)
    return res
'''